packages/integrations/markdoc/README.md

```diff
@@ -0,0 +1,342 @@
+});
+```
+
+Then, you can wire this render name (`'Aside'`) to a component from the `components` prop via the `<Content />` component:
+
+
+```astro
+---
+import { getEntryBySlug } from 'astro:content';
+import Aside from '../components/Aside.astro';
+
+const entry = await getEntryBySlug('docs', 'why-markdoc');
+const { Content } = await entry.render();
+---
+
+<Content
+  components={{
+    Aside: Aside,
+  }}
+/>
+```
+
+#### Render Markdoc nodes / HTML elements as Astro components
+
+You may also want to map standard HTML elements like headings and paragraphs to components. For this, you can configure a custom [Markdoc node][markdoc-nodes]. This example overrides Markdoc's `heading` node to render a `Heading` component, passing the built-in `level` attribute as a prop:
+
+```js
+// astro.config.mjs
+import { defineConfig } from 'astro/config';
+import markdoc from '@astrojs/markdoc';
+
+// https://astro.build/config
+export default defineConfig({
+	integrations: [
+		markdoc({
+			nodes: {
+				heading: {
+					render: 'Heading',
+          // Markdoc requires type defs for each attribute.
+          // These should mirror the `Props` type of the component
+          // you are rendering. 
+          // See Markdoc's documentation on defining attributes
+          // https://markdoc.dev/docs/attributes#defining-attributes
+          attributes: {
+            level: { type: String },
+          }
+				},
+			},
+		}),
+	],
+});
+```
+
+Now, you can map the string passed to render (`'Heading'` in this example) to a component import. This is configured from the `<Content />` component used to render your Markdoc using the `components` prop:
+
+```astro
+---
+import { getEntryBySlug } from 'astro:content';
+import Heading from '../components/Heading.astro';
+
+const entry = await getEntryBySlug('docs', 'why-markdoc');
+const { Content } = await entry.render();
+---
+
+<Content
+  components={{
+    Heading: Heading,
+  }}
+/>
+```
+
+📚 [Find all of Markdoc's built-in nodes and node attributes on their documentation.](https://markdoc.dev/docs/nodes#built-in-nodes)
+
+#### Use client-side UI components
+
+Today, the `components` prop does not support the `client:` directive for hydrating components. To embed client-side components, create a wrapper `.astro` file to import your component and apply a `client:` directive manually.
+
+This example wraps a `Aside.tsx` component with a `ClientAside.astro` wrapper:
+
+```astro
+---
+// src/components/ClientAside.astro
+import Aside from './Aside';
+---
+
+<Aside client:load />
+```
+
+This component [can be applied via the `components` prop](#render-html-elements-as-astro-components):
+
+```astro
+---
+// src/pages/why-markdoc.astro
+import { getEntryBySlug } from 'astro:content';
+import ClientAside from '../components/ClientAside.astro';
+
+const entry = await getEntryBySlug('docs', 'why-markdoc');
+const { Content } = await entry.render();
+---
+
+<Content
+  components={{
+    aside: ClientAside,
```

Comment:
Now I'm left wondering why this `aside` is lower case, when the others are upper case. Is this a typo? Were the others? (tbh, this is more what I would have expected!)

Reply:
Just a typo! Should be uppercase